.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,49 @@
+name: Integration Tests
+
+on: [pull_request, push]
+
+jobs:
+  fprime-integration:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8, 3.9]
+        fprime-version: [NASA-v1.5.0, NASA-v1.5.1, NASA-v1.5.2, NASA-v1.5.3, v2.0.0, devel]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .
+      - name: clone fprime and gps-tutorial
+        run: |
+          git clone -b ${{ matrix.fprime-version }} https://github.com/nasa/fprime.git
+          git clone https://github.com/fprime-community/gps-tutorial.git
+      - name: Revert gps-tutorial version if needed
+        if: |
+          matrix.fprime-version == 'NASA-v1.5.0' || 
+          matrix.fprime-version == 'NASA-v1.5.1' || 
+          matrix.fprime-version == 'NASA-v1.5.2' || 
+          matrix.fprime-version == 'NASA-v1.5.3'
+        run: |
+          cd gps-tutorial
+          git checkout 99cef07f4b43aebd4d3e9298081655b33be859b4
+          cd ..
+      - name: Test generation and building on Ref
+        run: |
+          cd fprime/Ref
```

Comment:
Instead of all this manual directory changing, github actions allows you to set a working directory. This might clean things up a little bit.

Ex:

```yaml
- name: Test generation and building on Ref
   working-directory: fprime/Ref
   run: |
          fprime-util generate
          fprime-util build
```